hypertrace-graphql-metric-schema/src/main/java/org/hypertrace/graphql/metric/request/DefaultMetricAggregationRequestBuilder.java

```diff
@@ -72,26 +74,27 @@
   }
 
   public MetricAggregationRequest build(
-      AttributeModel attribute,
+      AttributeAssociation<AttributeExpression> attributeExpression,
       AttributeModelMetricAggregationType aggregationType,
       List<Object> arguments) {
-    return this.build(attribute, aggregationType, arguments, false);
+    return this.build(attributeExpression, aggregationType, arguments, false);
   }
 
-  public MetricAggregationRequest build(
-      AttributeModel attribute,
+  private MetricAggregationRequest build(
+      AttributeAssociation<AttributeExpression> attributeExpression,
       AttributeModelMetricAggregationType aggregationType,
       List<Object> arguments,
       boolean baseline) {
-    return new DefaultMetricAggregationRequest(attribute, aggregationType, arguments, baseline);
+    return new DefaultMetricAggregationRequest(
+        attributeExpression, aggregationType, arguments, baseline);
   }
 
   private MetricAggregationRequest requestForAggregationField(
-      AttributeModel attribute, SelectedField field) {
+      AttributeAssociation<AttributeExpression> attributeExpression, SelectedField field) {
```

Comment:
I still think `attributeExpression` -> `attributeAssociation` would be more readable. It is getting confused with actual AttributeExpression and association with the model while reading. What do you think - should we rename?

Reply:
Done